### PR TITLE
CASSANDRA-19424 Check for expired certificate during start up validation

### PR DESCRIPTION
**Pull request by sarankk**

### CODE REVIEW

File: cassandra-analytics-core/src/main/java/org/apache/cassandra/spark/validation/KeyStoreValidation.java [hunk 1]
```diff
@@ -81,6 +85,15 @@ public void validate()
                 throw new RuntimeException("KeyStore is empty");
             }
 
+            for (Enumeration<String> aliases = keyStore.aliases(); aliases.hasMoreElements();)
+            {
+                Certificate cert = keyStore.getCertificate(aliases.nextElement());
+                if (cert instanceof X509Certificate && !(cert instanceof PemX509Certificate))
```
Comment (frankgh) [inline cassandra-analytics-core/src/main/java/org/apache/cassandra/spark/validation/KeyStoreValidation.java]: I don't think we need to worry about `PemX509Certificate`s.

File: cassandra-analytics-core/src/test/java/org/apache/cassandra/spark/validation/KeyStoreValidationTests.java [hunk 2]
```diff
@@ -97,4 +98,15 @@ public void testValidKeyStore()
...
+
+    @Test
+    public void testExpiredKeyStore()
+    {
+        SecretsProvider secrets = TestSecretsProvider.forKeyStore("PKCS12", "keystore-expired.p12", "qwerty");
+        KeyStoreValidation validation = new KeyStoreValidation(secrets);
+
+        Throwable throwable = validation.perform();
+        assertInstanceOf(RuntimeException.class, throwable);
+        assertThat(throwable.getMessage()).startsWith("Certificate expired, valid NotAfter: ");
```
Comment (frankgh) [inline cassandra-analytics-core/src/test/java/org/apache/cassandra/spark/validation/KeyStoreValidationTests.java]: the error message is odd, Should we just say:

`Certificate expired. NotAfter:` instead?

**Review by frankgh — COMMENTED** [review]
Looks good, I have a couple of minor comments

File: cassandra-analytics-core/src/test/java/org/apache/cassandra/spark/validation/KeyStoreValidationTests.java (hunk 2 above)
Comment (sarankk) [reply cassandra-analytics-core/src/test/java/org/apache/cassandra/spark/validation/KeyStoreValidationTests.java]: Updated to `Certificate expired. ` instead

File: cassandra-analytics-core/src/main/java/org/apache/cassandra/spark/validation/KeyStoreValidation.java [hunk 3]
```diff
@@ -81,6 +84,15 @@ public void validate()
...
             }
 
+            for (Enumeration<String> aliases = keyStore.aliases(); aliases.hasMoreElements();)
+            {
+                Certificate cert = keyStore.getCertificate(aliases.nextElement());
+                if (cert instanceof X509Certificate)
+                {
+                    ((X509Certificate) cert).checkValidity();
+                }
+            }
```
Comment (frankgh) [inline cassandra-analytics-core/src/main/java/org/apache/cassandra/spark/validation/KeyStoreValidation.java]: Can we collapse the for lops in lines 87 and 96:
```suggestion
            for (Enumeration<String> aliases = keyStore.aliases(); aliases.hasMoreElements();)
            {
                String alias = aliases.nextElement();
                Certificate cert = keyStore.getCertificate(alias);
                if (cert instanceof X509Certificate)
                {
                    ((X509Certificate) cert).checkValidity();
                }

                Key key = keyStore.getKey(alias, password);
                if (key != null && key instanceof PrivateKey)
                {
                    return;  // KeyStore contains a private key
                }
            }
```

**Review by frankgh — COMMENTED** [review]
One additional minor comment

File: cassandra-analytics-core/src/main/java/org/apache/cassandra/spark/validation/KeyStoreValidation.java (hunk 3 above)
Comment (frankgh) [reply cassandra-analytics-core/src/main/java/org/apache/cassandra/spark/validation/KeyStoreValidation.java]: ah, I just realized that the second loop looks for a PK< and it if finds it, it exits the validation as successful. Please disregard this comment.

**Review by frankgh — APPROVED** [review]
+1

**Review by yifan-c — APPROVED** [review]

**Pull request merged**